packages/core/src/config.ts

```diff
@@ -446,13 +454,12 @@ const composeFormatConfig = (format: Format): RsbuildConfig => {
             },
             module: {
               parser: {
-                javascript: {
-                  importMeta: false,
-                },
+                javascript: jsParserOptions,
               },
             },
             optimization: {
               concatenateModules: true,
+              sideEffects: 'flag',
```

Comment:
What is this for?

Reply:
https://webpack.js.org/configuration/optimization/#optimizationsideeffects, if a dep didn't marking `sideEffects: false` in package.json, then it will not be tree shaken.

**Why we need to change it to `'flag'`:**

to make the modules in the source code didn't get bail out from https://github.com/web-infra-dev/rspack/blob/main/crates/rspack_core/src/normal_module.rs#L525-L530 by when process concatenation. See the deleted code in snapshot.

**What does it cost?**

When bundling other modules, tree shaking performance may not be optimal. For cases sensitive to tree shaking, such as ESM, we typically avoid bundling the dependencies.

In the long run, it would be better to add a configuration that allows for fine-grained control over the level of tree shaking applied to specific modules.